src/basic/selinux-util.c

```diff
@@ -133,30 +133,38 @@ static int open_label_db(void) {
 int mac_selinux_init(void) {
 #if HAVE_SELINUX
         int r;
+        bool have_status_page = false;
 
         if (initialized)
                 return 0;
 
         if (!mac_selinux_use())
                 return 0;
 
-        r = selinux_status_open(/* no netlink fallback */ 0);
-        if (r < 0)
-                return log_enforcing_errno(errno, "Failed to open SELinux status page: %m");
+        r = selinux_status_open(/* netlink fallback */ 1);
+        if (r < 0) {
+                if (!ERRNO_IS_PRIVILEGE(errno))
+                        return log_enforcing_errno(errno, "Failed to open SELinux status page: %m");
+                log_warning_errno(errno, "selinux_status_open() failed, using the netlink fallback: %m");
```

Comment:
I think this is wrong: `selinux_status_open()` returns `-1` on error, `0` on success when the status page got mapped and `1` when mapping failed but the netlink socket got successfully opened.

So the `r < 0` branch should be simply:

```c
return log_enforcing_errno(errno, "Failed to open SELinux status page and connecting to SELinux netlink socket: %m");
```

Reply:
or was your intend to continue on error in permissive mode?

Reply:
Yes, it was my intent to continue. In Fedora selinux_status_open() fails with a permission error in both cases (with `0` and `1` as arguments).

Reply:
> `log_warning_errno(errno, "selinux_status_open() with fallback failed, not checking for policy reloads: %m");`

Thanks, I'll update it like that.
 
> Out of curiosity: with what daemons/CLI-applications are you seeing the permission errors (e.g. pid1 should already have access to the netlink socket, so the fallback should succeed.)

Most of them. logind, localed, networkd, and a bunch of others.